packages/arco-lib/src/generated/types/Form.ts

```diff
@@ -0,0 +1,73 @@
+import { Type } from "@sinclair/typebox";
+import { StringUnion } from '../../sunmao-helper';
+import { PRESET_PROPERTY_CATEGORY } from '@sunmao-ui/editor'
+import { TextPropertySchema } from "@sunmao-ui/runtime";
+
+export const FormControlPropsSchema = {
+    label: TextPropertySchema,
+    required: Type.Boolean({
+        title: 'Required',
+        category: PRESET_PROPERTY_CATEGORY.Basic
+    }),
+    hidden: Type.Boolean({
+        title: 'Hidden',
+        category: PRESET_PROPERTY_CATEGORY.Basic
+    }),
+    extra: Type.String({
+        title: 'Extra',
+        category: PRESET_PROPERTY_CATEGORY.Basic
+    }),
+    errorMsg: Type.String({
+        title: 'Error Message',
+        category: PRESET_PROPERTY_CATEGORY.Basic
+    }),
+    help: Type.String({
+        title: 'Help Message',
+        category: PRESET_PROPERTY_CATEGORY.Basic
+    }),
+    labelAlign: StringUnion(['left', 'right', 'unset'], {
+        title: 'Label Align',
+        category: PRESET_PROPERTY_CATEGORY.Layout
+    }),
+    colon: Type.Boolean({
+        title: 'Colon',
+        category: PRESET_PROPERTY_CATEGORY.Style
+    }),
+    labelCol: Type.Object({
+        span: Type.Number(),
+        offset: Type.Number()
+    }, {
+        title: 'Label Col',
+        category: PRESET_PROPERTY_CATEGORY.Layout
+    }),
+    wrapperCol: Type.Object({
+        span: Type.Number(),
+        offset: Type.Number()
+    }, {
+        title: 'Wrapper Col',
+        category: PRESET_PROPERTY_CATEGORY.Layout
+    })
+}
+
+export const FormPropsSchema = {
+    layout: StringUnion(['horizontal', 'vertical', 'inline'], {
+        title: 'Layout',
+        category: PRESET_PROPERTY_CATEGORY.Layout
+    }),
+    childrenLayout: StringUnion(['vertical', 'horizontal'], {
```

Comment:
What's the difference between `childrenLayout` and `layout`?

Reply:
I wanted to have each element in the `FormControl` on a separate line, but setting each `FormControl` individually would have been cumbersome, so I provided this property to be set uniformly in the `Form Stack`

Reply:
I wanted to have each element in the `FormControl` on a separate line, but setting each `FormControl` individually would have been cumbersome, so I provided this property to be set uniformly in the `Form Stack`

Reply:
I think that will be confusing. We'd better let the `FormControl` itselft to control its layout.